tensorflow_datasets/core/lazy_imports_lib_test.py

```diff
@@ -58,6 +58,12 @@ def test_bad_import(self):
     with self.assertRaisesWithPredicateMatch(ImportError, "extras_require"):
       _ = tfds.core.lazy_imports.test_foo
 
+  def test_lazy_import(self):
+    with self.assertRaisesWithPredicateMatch(ImportError, "extras_require"):
+      with tfds.core.lazy_importer():
+        import some_module
+
+      some_module.some_function()
```

Comment:
I believe we should only accept whitelisted deps to be registered. Accepting any deps may be more difficult for the user to debug if he make a typo. Instead we should add some `_ALLOWED_LAZY_DEPS = ['matplotlib', ...]`
```py
with lazy_imports():
  import matplotlib  # Works
  import invalid_import  # Raise 'Unknown import {name}. If you believe this is correct, please add it to the whitelisted imports'
```